ios_line: add`ios_line` resource module

earendilfr opened the pull request:
This module has been tested on:
- Cisco IOS-XE 17.06.05
- Cisco IOS-XE 16.12.6
- Cisco IOS 15.2(4)E10

##### SUMMARY
Add a new module `ios_line` to configure the lines part of Cisco IOS devices.
Currently, only line console and line vty are managed

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ios_line

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
The current module configure all the line console and vty.
It take in count the case that:
- we cannot remove the `line console 0`
- we cannot remove the `line vty` between 0 and 4

earendilfr committed:
ios_service: Initiale commit to create the module `ios_service`

This module has been tested on:
- Cisco IOS-XE 17.06.05

earendilfr committed:
ios_service: Initiale commit to create the module `ios_service`

This module has been tested on:
- Cisco IOS-XE 17.06.05

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

KB-perByte commented:
Hey @earendilfr, we would be actively reviewing this PR and plan to merge it in future releases.
Please note we need some integration tests with the module contribution. Let us know if you need any help.
Thank you for your contribution.
Regards

earendilfr commented:
> Hey @earendilfr, we would be actively reviewing this PR and plan to merge it in future releases. Please note we need some integration tests with the module contribution. Let us know if you need any help. Thank you for your contribution. Regards

Hey @KB-perByte, currently, it seems working fine from my point of view concerning this module.
Did you need other update concerning this PR ?

And I will also perform a PR on the repo https://github.com/ansible-network/resource_module_models top add this module too.

KB-perByte commented:
@earendilfr There needs to be some integration tests [here](https://github.com/ansible-collections/cisco.ios/tree/main/tests/integration/targets), covering bare minimum functionality of the PR.
Could you please add some?
Regards

earendilfr committed:
Initial commit for tests/integration

earendilfr committed:
Initial commit for tests/integration

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

earendilfr committed:
Fix some errors in tests/integration + remove initial space at startup

earendilfr committed:
Fix some errors in tests/integration + remove initial space at startup

earendilfr committed:
Add the missing part to take in count `aux` lines

earendilfr committed:
Add the missing part to take in count `aux` lines

earendilfr committed:
Fixe an error when we modify the length of the active line

Need to force to resend the command `terminal length 0` in this case

earendilfr committed:
Fixe an error when we modify the length of the active line

Need to force to resend the command `terminal length 0` in this case

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

earendilfr commented:
Potentially, I can also fix the issue #730: I have added a workaround to force to send the command `do terminal length 0` if I see that I have change the length configuration for one line

earendilfr committed:
Update the way to take in count the input ACL

Allow to take in count the case where the ACL should be put for the VRF

earendilfr committed:
Update the way to take in count the input ACL

Allow to take in count the case where the ACL should be put for the VRF

KB-perByte commented:
Hey @earendilfr, sorry for the delayed response, 
`ansible-test network-integration --python 3.11 --inventory /inventory ios_line -vvv`
You may use this command to check it against an ios xe appliance, note that the inventory host needs to be named `ios`.
Please, consider tagging me in conversations, it helps me get back faster.
Regards.

earendilfr committed:
- Fixe the integrations test to be correct

- Fixe the config module to:
 - doesn't modify lines not indicated with the deleted and replaced
   profiles
 - take correctly in case the case where we have aux line present on
   device

earendilfr committed:
- Fixe the integrations test to be correct

- Fixe the config module to:
 - doesn't modify lines not indicated with the deleted and replaced
   profiles
 - take correctly in case the case where we have aux line present on
   device

earendilfr commented:
> Hey @earendilfr, sorry for the delayed response, `ansible-test network-integration --python 3.11 --inventory /inventory ios_line -vvv` You may use this command to check it against an ios xe appliance, note that the inventory host needs to be named `ios`. Please, consider tagging me in conversations, it helps me get back faster. Regards.

Hi @KB-perByte ,
Thanks a lot : so, I have also write the integration tests and validate them.

I have also fixed something: like in ios_acls module, I have the `overridden`status to replace all configured lines and remove them not indicated; and the `replaced` state to replace configuration for the defined lines.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

KB-perByte committed:
Merge branch 'main' into new-ios_line

earendilfr committed:
Merge remote-tracking branch 'origin/new-ios_line' into new-ios_line

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

NilashishC committed:
Merge branch 'main' into new-ios_line

earendilfr committed:
Merge branch 'main' into new-ios_line
